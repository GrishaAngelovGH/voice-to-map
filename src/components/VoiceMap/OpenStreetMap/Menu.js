import BottomNavigation from '@mui/material/BottomNavigation'
import BottomNavigationAction from '@mui/material/BottomNavigationAction'

import HistoryIcon from '@mui/icons-material/History'
import DesktopWindowsIcon from '@mui/icons-material/DesktopWindows'

const Menu = () => {
    const handleChange = (event, index) => {

    }

    return (
        <div className="row g-0">
            <div className="col-md-12">
                <BottomNavigation showLabels={true} onChange={handleChange}>
                    <BottomNavigationAction label="History" icon={<HistoryIcon />} />
                    <BottomNavigationAction label="Workflow" icon={<DesktopWindowsIcon />} />
                </BottomNavigation>
            </div>
        </div>
    )
}

export default Menu