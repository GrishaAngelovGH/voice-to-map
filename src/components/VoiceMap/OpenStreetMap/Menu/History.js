import React from 'react'
import { useSelector } from 'react-redux'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'
import Slide from '@mui/material/Slide'

import RoomIcon from '@mui/icons-material/Room'
import CloseIcon from '@mui/icons-material/Close'

const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />
})

const History = ({ onClose }) => {
    const history = useSelector(state => state.locations.history)

    return (
        <Dialog fullScreen={true} open={true} TransitionComponent={Transition} data-testid="history">
            <AppBar sx={{ position: 'sticky' }}>
                <Toolbar>
                    <IconButton
                        edge="start"
                        color="inherit"
                        onClick={onClose}
                        aria-label="close"
                    >
                        <CloseIcon />
                    </IconButton>
                    <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
                        History
                    </Typography>
                </Toolbar>
            </AppBar>
            <List>
                {
                    history.map((v, i) => (
                        <Fragment key={v.place_id}>
                            <ListItem button>
                                <ListItemText primary={<span>{`#${i + 1}`} <RoomIcon /> {v.display_name}</span>} />
                            </ListItem>
                            <Divider />
                        </Fragment>
                    ))
                }
            </List>
        </Dialog>
    )
}

export default History