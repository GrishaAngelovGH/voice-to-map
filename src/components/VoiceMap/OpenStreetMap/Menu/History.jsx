import React from 'react'
import { useSelector } from 'react-redux'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'

import RoomIcon from '@mui/icons-material/Room'
import CloseIcon from '@mui/icons-material/Close'

const History = ({ onClose }) => {
	const history = useSelector(state => state.locations.history)

	return (
		<Dialog
			fullScreen={true}
			open={true}
		>
			<AppBar sx={{ position: 'sticky' }}>
				<Toolbar>
					<IconButton
						edge='start'
						color='inherit'
						onClick={onClose}
						aria-label='close'
					>
						<CloseIcon />
					</IconButton>
					<Typography sx={{ ml: 2, flex: 1 }} variant='h6' component='div'>
						History
					</Typography>
				</Toolbar>
			</AppBar>
			<List>
				{
					history.map((v, i) => (
						<Fragment key={v.place_id}>
							<ListItem>
								<ListItemText primary={<span>{`#${i + 1}`} <RoomIcon /> {v.display_name}</span>} />
							</ListItem>
							<Divider />
						</Fragment>
					))
				}
			</List>
		</Dialog>
	)
}

export default History