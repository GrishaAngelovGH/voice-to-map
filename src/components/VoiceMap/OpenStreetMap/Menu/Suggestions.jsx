import React from 'react'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'
import Slide from '@mui/material/Slide'

import RecordVoiceOverIcon from '@mui/icons-material/RecordVoiceOver'
import CloseIcon from '@mui/icons-material/Close'

const Transition = React.forwardRef(function Transition(props, ref) {
	return <Slide direction='up' ref={ref} {...props} />
})

const suggestions = [
	'Bulgaria Sofia National Palace of Culture',
	'Bulgaria Sofia Zoo',
	'Bulgaria Sofia National History Museum',
	'London Palace of Westminster',
	'London Buckingham Palace',
	'London Piccadilly Circus',
	'London Big Ben',
	'Tower Bridge London',
	'Germany Brandenburg Gate',
	'France Eiffel Tower'
]

const Suggestions = ({ onClose }) => (
	<Dialog fullScreen={true} open={true} TransitionComponent={Transition}>
		<AppBar sx={{ position: 'sticky' }} color='secondary'>
			<Toolbar>
				<IconButton
					edge='start'
					color='inherit'
					onClick={onClose}
					aria-label='close'
				>
					<CloseIcon />
				</IconButton>
				<Typography sx={{ ml: 2, flex: 1 }} variant='h6' component='div'>
					Voice Search Suggestions
				</Typography>
			</Toolbar>
		</AppBar>
		<List>
			{
				suggestions.map((v, i) => (
					<Fragment key={i}>
						<ListItem button>
							<Fragment >
								<ListItemText primary={<span><RecordVoiceOverIcon /> {v}</span>} />
							</Fragment>
						</ListItem>
						<Divider />
					</Fragment>
				))
			}
		</List>
	</Dialog>
)

export default Suggestions
