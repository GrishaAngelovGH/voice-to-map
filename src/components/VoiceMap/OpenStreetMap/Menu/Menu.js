import { useState } from 'react'

import BottomNavigation from '@mui/material/BottomNavigation'
import BottomNavigationAction from '@mui/material/BottomNavigationAction'
import Drawer from '@mui/material/Drawer'

import ListIcon from '@mui/icons-material/List'
import HistoryIcon from '@mui/icons-material/History'
import DesktopWindowsIcon from '@mui/icons-material/DesktopWindows'

import History from './History'

const Menu = ({ disableHistory }) => {
    const [showHistory, setShowHistory] = useState(false)
    const [showSuggestions, setShowSuggestions] = useState(false)
    const [showWorkflow, setShowWorkflow] = useState(false)

    const handleChange = (event, index) => {
        if (index === 0) {
            !disableHistory && setShowHistory(true)
            return
        }

        index === 1 ? setShowSuggestions(true) : setShowWorkflow(true)
    }

    const handleClose = () => {
        setShowHistory(false)
        setShowSuggestions(false)
        setShowWorkflow(false)
    }

    return (
        <div className="row g-0">
            <div className="col-md-12">
                <BottomNavigation showLabels={true} onChange={handleChange}>
                    <BottomNavigationAction label="History" icon={<HistoryIcon />} />
                    <BottomNavigationAction label="Suggestions" icon={<ListIcon />} />
                    <BottomNavigationAction label="Workflow" icon={<DesktopWindowsIcon />} />
                </BottomNavigation>

                <Drawer open={showHistory || showWorkflow || showSuggestions} anchor={'top'} onClose={handleClose}>
                    {showHistory && <History onClose={handleClose} />}
                    {showSuggestions && <div>SUGGESTIONS</div>}
                    {showWorkflow && <div>WORKFLOW</div>}
                </Drawer>
            </div>
        </div>
    )
}

export default Menu