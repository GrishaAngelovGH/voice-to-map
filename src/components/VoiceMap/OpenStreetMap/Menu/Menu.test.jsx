import { render } from '@testing-library/react'

import Menu from './Menu'

test('should render Menu component', () => {
	const view = render(<Menu />)

	expect(view).toMatchSnapshot()
})