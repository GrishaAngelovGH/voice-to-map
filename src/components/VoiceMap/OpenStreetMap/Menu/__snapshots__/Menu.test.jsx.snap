// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Menu component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0"
      >
        <div
          class="col-md-12"
        >
          <div
            class="MuiBottomNavigation-root css-7ax0nd-MuiBottomNavigation-root"
          >
            <button
              class="MuiButtonBase-root MuiBottomNavigationAction-root css-1scjuq2-MuiButtonBase-root-MuiBottomNavigationAction-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="HistoryIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9m-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label css-1kcvi62-MuiBottomNavigationAction-label"
              >
                History
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiBottomNavigationAction-root css-1scjuq2-MuiButtonBase-root-MuiBottomNavigationAction-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ListIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h2v-2H3zm0 4h2v-2H3zm0-8h2V7H3zm4 4h14v-2H7zm0 4h14v-2H7zM7 7v2h14V7z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label css-1kcvi62-MuiBottomNavigationAction-label"
              >
                Suggestions
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0"
    >
      <div
        class="col-md-12"
      >
        <div
          class="MuiBottomNavigation-root css-7ax0nd-MuiBottomNavigation-root"
        >
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root css-1scjuq2-MuiButtonBase-root-MuiBottomNavigationAction-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="HistoryIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9m-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label css-1kcvi62-MuiBottomNavigationAction-label"
            >
              History
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root css-1scjuq2-MuiButtonBase-root-MuiBottomNavigationAction-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ListIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h2v-2H3zm0 4h2v-2H3zm0-8h2V7H3zm4 4h14v-2H7zm0 4h14v-2H7zM7 7v2h14V7z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label css-1kcvi62-MuiBottomNavigationAction-label"
            >
              Suggestions
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
